actions/create_alert.py

```diff
@@ -11,16 +11,57 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
+import re
 
 from lib.actions import OpsGenieBaseAction
 
 
+def identify_entity(identity):
+    """
+    Detect the type of OpsGenie entity being processed: an id, username or name.
+
+    identity: string : Expects to be a string containing an OpsGenie entity and
+                       it's type separated by a single '-'
+
+    return: A dictionary with the form and type of the entity or None if the identity
+            can't be processed.
+    """
+    res = identity.rsplit("-", 1)
+    if len(res) != 2:
+        print("Skipping {} because it's not formed as '<identiy>-<type>'".format(identity))
+        return None
+
+    form, type_ = res
+    # Detect email form: "trinity@opsgenie.com"
+    if type_ == "user" and re.match(r"^[^@]+@[^@]+$", form):
```

Comment:
Would be more performant to compile these regexes and use the compiled patterns to match, especially since this is called in a loop.